sdk/eventhub/azure-eventhubs/azure/eventhub/_utils.py

```diff
@@ -0,0 +1,111 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from __future__ import unicode_literals
+
+import sys
+import platform
+
+from uamqp import types  # type: ignore
+from uamqp.message import MessageHeader  # type: ignore
+
+from azure.core.settings import settings # type: ignore
+
+from azure.eventhub import __version__
+from ._constants import PROP_PARTITION_KEY_AMQP_SYMBOL, MAX_USER_AGENT_LENGTH
+
+
+def create_properties(user_agent=None):
+    """
+    Format the properties with which to instantiate the connection.
+    This acts like a user agent over HTTP.
+
+    :rtype: dict
+    """
+    properties = {}
+    product = "azsdk-python-eventhubs"
+    properties[types.AMQPSymbol("product")] = product
+    properties[types.AMQPSymbol("version")] = __version__
+    framework = "Python {}.{}.{}, {}".format(
+        sys.version_info[0], sys.version_info[1], sys.version_info[2], platform.python_implementation()
+    )
+    properties[types.AMQPSymbol("framework")] = framework
+    platform_str = platform.platform()
+    properties[types.AMQPSymbol("platform")] = platform_str
+
+    final_user_agent = '{}/{} ({}, {})'.format(product, __version__, framework, platform_str)
+    if user_agent:
+        final_user_agent = '{}, {}'.format(final_user_agent, user_agent)
+
+    if len(final_user_agent) > MAX_USER_AGENT_LENGTH:
```

Comment:
Do we have any idea of our own size in all possible plateform by default (if no user agent is provided) compared to the max?
I'm curious to see if we're like 1000 characters large, or 10.

Reply:
Checked with @annatisch 512 is plenty